engine/src/main/scala/cromwell/services/MetadataServiceActor.scala

```diff
@@ -0,0 +1,60 @@
+package cromwell.services
+
+import akka.actor.{Actor, Props}
+import com.typesafe.config.Config
+import cromwell.backend.BackendJobDescriptorKey
+import cromwell.core.WorkflowId
+import cromwell.services.MetadataServiceActor._
+import cromwell.services.ServiceRegistryActor.ServiceRegistryMessage
+import org.joda.time.DateTime
+
+import scala.language.postfixOps
+
+object MetadataServiceActor {
+
+  final val MetadataServiceName = "MetadataService"
+
+  case class MetadataJobKey(callFqn: String, index: Option[Int], attempt: Int)
+  object MetadataJobKey {
+    def apply(backendJobKey: BackendJobDescriptorKey): MetadataJobKey = MetadataJobKey(backendJobKey.call.fullyQualifiedName, backendJobKey.index, backendJobKey.attempt)
+  }
+
+  case class MetadataKey(workflowId: WorkflowId, jobKey: Option[MetadataJobKey], key: String)
```

Comment:
Hmm, an ADT around an Option of an exiting ADT seems overkill to me for now, but I'll happily eat these words if it becomes a problem later